UPGRADE-3.0.md

```diff
@@ -0,0 +1,15 @@
+## UPGRADE from 2.x to 3.0
+
+### General
+
+* The `php-http/httplug` dependency requires is bumped to minimum ^2.1.
+* A client implementing `psr/http-client-implementation` is required.
+  To upgrade your application (default install) switch from guzzle 6 to guzzle 7, see the install instructions in the [README file](README.md)
+* All previous deprecated code in version 2 is removed.
+
+### Authetication methods
+
+* `Github\Client::AUTH_URL_TOKEN` use `Github\Client::AUTH_ACCESS_TOKEN` instead.
```

Comment:
I have always found it kinda odd that the authentication plugin should know about the client and it's constants. Maybe we should move these constants to the auth plugin, or somewhere neutral?

Reply:
Good idea, I've created an issue for this. Let's keep it like this for now so we can release 3.0 and prepare it for 4.0!